llvm/lib/Target/EraVM/EraVMInstrInfo.td

```diff
@@ -993,7 +993,6 @@ def JCs: IJump<OpJump, SrcStackAbsolute, (ins stackin:$dest), "$dest", []>;
 let isTerminator = 1, isBranch = 1, isBarrier = 1 in {
 def J: Pseudo<(outs), (ins jmptarget:$dest), [(EraVMbrcc bb:$dest, 0)]>,
        PseudoInstExpansion<(JC jmptarget:$dest, 0)>;
-def J_s: Pseudo<(outs), (ins stackin:$dest), []>;
```

Comment:
In some optimizations, we utilize the implicit Flags register to identify the defs and uses of it. With this change, we would always assume that `JCs` instruction is using flags, which could be incorrect (as it is shown in the test). Even thought I don't think this change currently creates an issue, my suggestion is to keep this instruction, and think about how to remove this overhead later on. Wdyt?

Reply:
Finally had to introduce a separate commit https://github.com/matter-labs/era-compiler-llvm/pull/448/commits/fbdba35e29de815a70f550c76e85aded75635b2b and fold the expansion into https://github.com/matter-labs/era-compiler-llvm/pull/448/commits/3090f94936e71b1ea18274d1fa73419ce6e9f674, so J_s is properly expanded at any point in history (though, it seems integration tests are only executed for the HEAD commit of the branch).